backend/Controllers/cart.js

```diff
@@ -0,0 +1,66 @@
+const Cart = require("../Database/cart");
+const _ = require('lodash');
+
+exports.getCartItems = async (req,res) => {
+    let {cart} = req
+    try {
+        cart = await cart.populate('cartItems.product').execPopulate()
+        res.json({success:true, response: cart})
+    }
+    catch(error) {
+        res.json({success: false, response: error.message})
+    }
+}
+
+exports.addCartItems = async (req,res) => {
+    let {product} = req;
+    let {cart} = req;
+    try {
+        if(!cart.cartItems.id(product._id)){
+            cart = _.extend(cart, {cartItems: _.concat(cart.cartItems,{_id:product._id, product: product._id, quantity: 1})})
+            cart = await cart.save()
+            cart = await cart.populate('cartItems.product').execPopulate()
+            res.json({success:true, response: cart})
+        }
+        else {
+            res.json({success:true, response: "already exists in cart"})
+        }
+    }
+    catch(error) {
+        res.json({success: false, response: error.message});
+    }
+}
+
+exports.updateQuantityOfCartItems = async (req, res) => {
+    let cart = req.cart;
+    const { productId } = req.params;
+    const {quantity} = req.body;
+    let updateCartItem = cart.cartItems.id(productId)
+    updateCartItem = _.extend(updateCartItem,{quantity: quantity})
```

Comment:
Hi Sruthi , I must say the code is well written and structure, while reading it is found this issue if productId not present in cart items then it returns null. that would break the server or raise an issue.

Reply:
can you please this? As far as i have understood, you can only update a id that is in cart, in that case, there will always be a productId. Please correct if i have gotten it wrong